### PR TITLE
feat(web-components): adds support for form reset event

### PR DESCRIPTION
**Pull request by ad9242**
## Summary of the changes
Input components now reset to initial state when used on a form & a form reset is triggered.
Adds test coverage for components too.
Fixes issue in switch component where switch did not update correctly on toggle when showState & checked props were set

## Related issue
#80 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

File: packages/web-components/src/components/ic-radio-option/ic-radio-option.tsx [hunk 1]
```diff
@@ -76,10 +81,14 @@ export class RadioOption {
   }
 
   private defaultRadioValue: string = "";
+  private skipFocus = false;
```
Comment (jd239) [inline packages/web-components/src/components/ic-radio-option/ic-radio-option.tsx:84]: Just so I understand better, why do we have a skipFocus flag? Was there initial a focus state on the radio button when reset was pressed?

File: packages/web-components/src/components/ic-radio-option/ic-radio-option.tsx [hunk 2]
```diff
@@ -37,7 +40,7 @@ export class RadioOption {
   /**
    * Provide a value for the input.
    */
-  @Prop() value!: string;
+  @Prop({ mutable: true }) value!: string;
```
Comment (jd239) [inline packages/web-components/src/components/ic-radio-option/ic-radio-option.tsx:43]: Does this need to be `mutable` as I cannot see where we are setting the value internally again?

**Review by jd239 — COMMENTED** [review]
Only two questions and not really code changes. Looks good! :)

File: packages/web-components/src/components/ic-radio-option/ic-radio-option.tsx (hunk 1 above)
Comment (ad9242) [reply packages/web-components/src/components/ic-radio-option/ic-radio-option.tsx:84]: We are changing the state of selected which invokes the handleClick by way of the watch on the selected prop. When this happens the radio receives focus, but we do not want it to in this case

File: packages/web-components/src/components/ic-radio-option/ic-radio-option.tsx (hunk 2 above)
Comment (ad9242) [reply packages/web-components/src/components/ic-radio-option/ic-radio-option.tsx:43]: Was giving a warning in the unit tests, as the value is updated in handleClick method

**Review by jd239 — APPROVED** [review]

**Review by GCHQ-Developer-741 — APPROVED** [review]

**Pull request merged**